plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/actions/RequestModelActionHandler.java

```diff
@@ -29,18 +30,23 @@
    @Inject
    protected ModelFactory modelFactory;
 
+   @Inject
+   private ModelSourceWatcher modelSourceWatcher;
+
    @Override
    public List<Action> executeAction(final RequestModelAction action, final GModelState modelState) {
-
       GModelRoot model = modelFactory.loadModel(action, modelState);
       modelState.setRoot(model);
       modelState.setClientOptions(action.getOptions());
+
+      modelSourceWatcher.startWatching(modelState);
```

Comment:
It seems a bit asymmetrical that we start the watching here but we never actually stop the watching but do so by having the watcher listen to the client session. I know it is like this because the client options are only sent through the request model action so this is the earliest we can actually start the watching. We could extend the ClientSessionManager with an 'updateClientSession' or something similar so that the whole file watcher could be an encapsulated class. But this is just a random thought and does not have to be done in this PR ;-)

Reply:
Right, I agree! Hm, but I feel this needs more thought. So I'd rather prefer doing that in a follow up PR.